src/init.cpp

```diff
@@ -1078,7 +1086,7 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
 
     // Start the lightweight task scheduler thread
     CScheduler::Function serviceLoop = boost::bind(&CScheduler::serviceQueue, &scheduler);
-    threadGroup.create_thread(boost::bind(&TraceThread<CScheduler::Function>, "scheduler", serviceLoop));
+    scheduler_thread = std::thread(std::bind(&TraceThread<CScheduler::Function>, "scheduler", serviceLoop));
```

Comment:
Github weirdly formatted my comment so you couldn't see the lambda.

    scheduler_thread = std::thread([&]{
        TraceThread("scheduler", [&]{scheduler.serviceQueue();} );
     };

Does something like that not work? Maybe it's not much better than bind...